### DISCUSSION TITLE
DNS server distribution to device

### DISCUSSION THREAD

**Discussion by typepub** [openthread/openthread#9651] [Q&A] [answered]
Hello everyone, 

I want my Thread device to get DNS servers when connecting to the network. I understand that [SRP can be used with mDNS to advertise services](https://openthread.io/codelabs/openthread-border-router?hl=en#1) but it doesn't suit y need (or maybe I misunderstood something?). 

What is the preferred method to achieve that please? I didn't found documentation about that. 

Thanks.

**Comment by jwhui** [comment, marked as answer]
OpenThread does implement DNS recursive resolver. Check out:
- https://github.com/openthread/openthread/pull/8668
- https://github.com/openthread/ot-br-posix/pull/1777
